packages/gestalt/src/RadioGroupButton.js

```diff
@@ -0,0 +1,192 @@
+
+  const handleHover: (isHovered: boolean) => void = (isHovered: boolean) => setHover(isHovered);
+
+  let borderStyle = styles.Border;
+  if (disabled && checked) {
+    borderStyle = styles.BorderDisabledChecked;
+  } else if (!disabled && checked) {
+    borderStyle = styles.BorderSelected;
+  } else if (!disabled && hovered) {
+    borderStyle = styles.BorderHovered;
+  }
+
+  let borderWidth = styles.BorderUnchecked;
+  if (disabled && !checked) {
+    borderWidth = styles.BorderDisabled;
+  } else if (checked && size === 'sm') {
+    borderWidth = styles.BorderCheckedSm;
+  } else if (checked && size === 'md') {
+    borderWidth = styles.BorderCheckedMd;
+  }
+
+  const styleSize = size === 'sm' ? controlStyles.sizeSm : controlStyles.sizeMd;
+
+  const bgStyle = disabled && !checked ? styles.BgDisabled : styles.BgEnabled;
+
+  const { isFocusVisible } = useFocusVisible();
+
+  const { parentName } = useRadioGroupContext();
+  if (parentName !== 'RadioGroup') {
+    throw new Error(`RadioGroup.RadioButton must be used within a RadioGroup.`);
+  }
+
+  return (
+    <Box
+      alignItems={subtext || image ? 'start' : 'center'}
+      display="flex"
+      justifyContent="start"
+      marginStart={-1}
+      marginEnd={-1}
+    >
+      <Label htmlFor={id}>
+        <Box paddingX={1}>
+          <div
+            className={classnames(
+              bgStyle,
+              borderStyle,
+              borderWidth,
+              styleSize,
+              styles.RadioButton,
+              {
+                [focusStyles.accessibilityOutlineFocus]: focused && isFocusVisible,
+              },
+            )}
+          >
+            <input
+              checked={checked}
+              className={classnames(controlStyles.input, styleSize, {
+                [styles.InputEnabled]: !disabled,
+              })}
+              disabled={disabled}
+              id={id}
+              name={name}
+              onBlur={handleBlur}
+              onChange={handleChange}
+              onFocus={handleFocus}
+              onMouseEnter={() => handleHover(true)}
+              onMouseLeave={() => handleHover(false)}
+              ref={ref}
+              type="radio"
+              value={value}
+            />
+          </div>
+        </Box>
+      </Label>
+      {Boolean(image) && <Box paddingX={1}>{image}</Box>}
+      {label && (
+        <Label htmlFor={id}>
+          <Box paddingX={1}>
+            <Text color={disabled ? 'subtle' : undefined} size={size === 'sm' ? '200' : '300'}>
+              {label}
+            </Text>
+            {subtext && (
+              <Box paddingY={1}>
+                <Text color="subtle" size={size === 'sm' ? '200' : '300'}>
+                  <Box display="visuallyHidden">:</Box> {subtext}
+                </Text>
+              </Box>
+            )}
+          </Box>
+        </Label>
+      )}
+    </Box>
+  );
+});
+
+RadioGroupButtonWithForwardRef.displayName = 'RadioGroupButton';
```

Comment:
The displayName should match the exported (from Gestalt) sub-component name, otherwise dev tools markup is confusing